adapters/adgeneration/adgeneration_test.go

```diff
@@ -174,3 +175,82 @@ func TestCreateAd(t *testing.T) {
+	}
+	// default Currency InternalRequest
+	defaultCurBidderRespoonse, errs := bidder.MakeBids(internalRequest, &externalRequest, &response)
+	if len(errs) > 0 {
+		t.Errorf("MakeBids return errors. errors: %v", errs)
+	}
+	checkBidResponse(t, defaultCurBidderRespoonse, bidder.defaultCurrency)
+
+	// Specified Currency InternalRequest
+	usdCur := "USD"
+	internalRequest.Cur = []string{usdCur}
+	specifiedCurBidderResponse, errs := bidder.MakeBids(internalRequest, &externalRequest, &response)
+	if len(errs) > 0 {
+		t.Errorf("MakeBids return errors. errors: %v", errs)
+	}
+	checkBidResponse(t, specifiedCurBidderResponse, usdCur)
+
+}
+
+func checkBidResponse(t *testing.T, bidderRespoonse *adapters.BidderResponse, expectedCurrency string) {
+
+	var expectedAdM string = "<div id=\"medibasspContainer\">\n<iframe src=\"https://dummy-iframe.com></iframe>\n</div>\n<img src=\"https://dummy-beacon.com\">"
+	var expectedID string = "58278"
+	var expectedImpID = "bidRequest-success-test"
+	var expectedPrice float64 = 30.0
+	var expectedW uint64 = 300
+	var expectedH uint64 = 250
+	var expectedCrID string = "Dummy_supership.jp"
+	var extectedDealID string = "test-deal-id"
+
+	// Currency
+	if bidderRespoonse.Currency != expectedCurrency {
+		t.Errorf("Currency %v does not equal %v", bidderRespoonse.Currency, expectedCurrency)
+	}
+	// Bids
+	if len(bidderRespoonse.Bids) != 1 {
```

Comment:
If you don't move to json tests, consider using the testify package to simplify these lines to:
```
assert.Equal(t, 1, len(bidderRespoonse.Bids), "Bid Responses")
assert.Equal(t, expectedID, bidderRespoonse.Bids[0].Bid.ID, "Bid ID")
...
```
The  testify package will auto generate useful error messages.